lib/index.js

```diff
@@ -4,14 +4,20 @@
 'use strict';
 
 const bedrock = require('bedrock');
-const {config} = bedrock;
+const {config, util: {BedrockError}} = bedrock;
 const https = require('https');
 
 // module API
 const api = {};
-module.exports = api;
 
-api.httpsAgent = api.agent = null;
+module.exports = new Proxy(api, {
```

Comment:
I don't think this is right. I think we want to wrap `httpsAgent` instead ... I'll look around for a pattern.

Reply:
I think we need something more like this:

https://gist.github.com/dlongley/b34ab693e531dc892c9ac3ebed1cc6b7

Reply:
How is this different better?

Reply:
What we have doesn't actually work. And the tests don't test it properly.